src/server/cluster/cluster_slot_migration.cc

```diff
@@ -82,4 +101,69 @@ ClusterSlotMigration::Info ClusterSlotMigration::GetInfo() const {
   return {ctx.host, ctx.port, state_};
 }
 
+void ClusterSlotMigration::MainMigrationFb() {
+  VLOG(1) << "Main migration fiber started";
+
+  state_ = ClusterSlotMigration::C_FULL_SYNC;
+
+  // TODO add reconnection code
+  if (auto ec = InitiateDflyFullSync(); ec) {
+    LOG(WARNING) << "Error syncing with " << server().Description() << " " << ec << " "
+                 << ec.message();
+  }
+}
+
+std::error_code ClusterSlotMigration::InitiateDflyFullSync() {
+  shard_flows_.resize(source_shards_num_);
+  for (unsigned i = 0; i < source_shards_num_; ++i) {
+    shard_flows_[i].reset(new ClusterShardMigration(server(), i, sync_id_));
+  }
+
+  // Switch to new error handler that closes flow sockets.
+  auto err_handler = [this](const auto& ge) mutable {
+    // Make sure the flows are not in a state transition
+    lock_guard lk{flows_op_mu_};
+
+    // Unblock all sockets.
+    DefaultErrorHandler(ge);
+    for (auto& flow : shard_flows_)
+      flow->Cancel();
+  };
+  RETURN_ON_ERR(cntx_.SwitchErrorHandler(std::move(err_handler)));
+
+  std::atomic_uint32_t synced_shards = 0;
+  auto partition = Partition(source_shards_num_);
+  auto shard_cb = [&](unsigned index, auto*) {
+    for (auto id : partition[index]) {
+      auto ec = shard_flows_[id]->StartSyncFlow(&cntx_);
+      if (!ec) {
+        ++synced_shards;
+      } else {
+        cntx_.ReportError(ec);
+      }
+    }
+  };
+  // Lock to prevent the error handler from running instantly
+  // while the flows are in a mixed state.
+  lock_guard lk{flows_op_mu_};
+  shard_set->pool()->AwaitFiberOnAll(std::move(shard_cb));
+
+  VLOG(1) << synced_shards << " from " << source_shards_num_ << " shards were set flow";
+  if (synced_shards != source_shards_num_) {
+    cntx_.ReportError(std::make_error_code(errc::state_not_recoverable),
+                      "incorrect shards num, only for tests");
+  }
+
+  RETURN_ON_ERR(cntx_.GetError());
+
+  string request = absl::StrCat("DFLYMIGRATE SYNC ", sync_id_);
+
+  VLOG(1) << "Sending: " << request;
+  RETURN_ON_ERR(SendCommandAndReadResponse(request));
+
+  PC_RETURN_ON_BAD_RESPONSE(CheckRespIsSimpleReply("OK"));
+
```

Comment:
can JoinAllFlow is missing right?

Reply:
we do it in Destructor, I don't see why it should be here. Also, we plan not to sync flows so they can work independently, so I think joining here is superfluous,

Reply:
I think you will need a JoinAllFlow call, but probably no in this function but in the MainMigrationFb
f.e if there was error in one of the flows you will have cancel called for all flows and then you will want to join all flows before you reconnect.
Anyway we can add it in a later PR

Reply:
It looks we have some misunderstanding because in this we need SYNC command from my point of view